tests/scripts/analyze_outcomes.py

```diff
@@ -159,24 +166,32 @@ def read_outcome_file(outcome_file):
 
 def do_analyze_coverage(outcome_file, args):
     """Perform coverage analysis."""
+    log = TestLog()
+    log.info("\n*** Analyze coverage ***\n")
     outcomes = read_outcome_file(outcome_file)
-    Results.log("\n*** Analyze coverage ***\n")
-    results = analyze_outcomes(outcomes, args)
-    return results.error_count == 0
+    log = analyze_outcomes(log, outcomes, args)
+    return log
 
 def do_analyze_driver_vs_reference(outcome_file, args):
     """Perform driver vs reference analyze."""
-    execute_reference_driver_tests(args['component_ref'], \
-                                    args['component_driver'], outcome_file)
+    log = TestLog()
+
+    log = execute_reference_driver_tests(log, args['component_ref'], \
+                                         args['component_driver'], outcome_file)
+    if log.error_count != 0:
+        return log
 
     ignored_suites = ['test_suite_' + x for x in args['ignored_suites']]
 
     outcomes = read_outcome_file(outcome_file)
-    Results.log("\n*** Analyze driver {} vs reference {} ***\n".format(
+
+    log.info("\n*** Analyze driver {} vs reference {} ***\n".format(
```

Comment:
Note: I think this should be the first thing logged by the function, before calling `execute_reference_driver_tests`, in order to contextualize its output.

Reply:
Ok for the 1st comment.
As for the 2nd one, instead, I did it this way in order to mimic the behavior of `do_analyze_coverage`. In this way each test function, `do_analyze_coverage` or `do_analyze_driver_vs_reference` (or others in the future), creates its own log and it puts everything there. I thought that this way could provide a more self-contained output.
Wdyt?

Reply:
I think it depends if each `do_xxx` function has their own instance of `Results`/`TestLog` or if there's a long-lived instance in the main function, as Gilles suggests.